sim/core/spell_scaling_config.go

```diff
@@ -0,0 +1,19 @@
+package core
+
+type SpellScalingConfig struct {
+	BaseWeapon_Pct    float64
+	Coefficient       float64
+	EffectPerLevel    float64
+	BaseSpellLevel    int32
+	MaxSpellLevel     int32
+	ClassSpellScaling float64
+}
+
+func (config *SpellScalingConfig) CalcSpellDamagePct() float64 {
+
+	return config.BaseWeapon_Pct + (config.EffectPerLevel * (float64(config.MaxSpellLevel - config.BaseSpellLevel)) / 100)
+}
+
+func (config *SpellScalingConfig) CalcAddedSpellDamage() float64 {
+	return config.Coefficient * config.ClassSpellScaling
+}
```

Comment:
This is specific to the Spell Effect for `WEAPON_PERCENT_DAMAGE` and should probably not be here or if you want it to be part of core name it something that is clear its for this effect and not general `SpellScalingConfig`. You can move it to the warrior package if not.

Reply:
Moved to Warrior Package and renamed it for clarity.